### ISSUE TITLE
Mac Not Loading

### ISSUE THREAD

**Issue by ClawsForHands** [blushiemagic/tModLoader#400]
### Description

After downloading tmodloader and starting terraria, the Re-Logic screen shows up and fades to black, then the spinning beachball on the cursor comes up and the game doesn't load/is frozen indefinitely until I force quit it.

### Steps to Reproduce

1. Download and install tmodloader using the .jar (or manually, I've tried both)
2. Try to start terraria 

### Expected behavior 

Terraria doesn't start

### Actual behavior

Terraria should start normally

### Reproduction frequency

100%

### Versions

tModloader v0.10.5.3-v0.10

### Mods

none

### Additional Information

I have tried reinstalling terraria multiple times by deleting all terraria files, I know I'm installing it correctly, and it doesn't seem like others have this problem because theres nothing about it on google...

**Comment by stuff50** [comment]
After playing around wit this today I was able to get i to work. I held down the option key and clicked on the Terraria icon and it opened with no issue.

**Comment by ClawsForHands** [comment]
Oh so holding down the option key made it work? How weird! I’ll definetely try that once I’m back on my laptop tomorrow and reply back to this thread, thanks!

**Comment by stuff50** [comment]
Well using the option key worked for about two days, I'm back to a black screen again.

**Comment by JavidPack** [comment]
We are aware, we just still have no answers. None of us use Mac so we're reliant on Mac users with some debugging skills to debug this sort of thing. Running the game from the command line might shed some light on things, but I don't know the command to do that on Mac.

**Comment by Chicken-Bones** [comment]
Yeah, unfortunately there's no OSX developer on the team. Mac support is predominantly just linux support with the occasional tweak. Very sorry that we can't do anything about it at the moment.

**Comment by Jofairden** [comment]
Maybe one of us can virtualboot OSX and dev on there :P

**Comment by jtnix** [comment]
Happening here also, OS X 12 (Sierra)
Console is reporting this before black screen / beachball:

> default	13:02:30.595813 -0400	Terraria.bin.osx	WARNING: The Gestalt selector gestaltSystemVersion is returning 10.9.6 instead of 10.12.6. This is not a bug in Gestalt -- it is a documented limitation. Use NSProcessInfo's operatingSystemVersion property to get correct system version number.
> Call location:
> default	13:02:30.595938 -0400	Terraria.bin.osx	0   CarbonCore                          0x00007fff7b86f1b1 ___Gestalt_SystemVersion_block_invoke + 121
> default	13:02:30.595959 -0400	Terraria.bin.osx	1   libdispatch.dylib                   0x00007fff900bd8fc _dispatch_client_callout + 8
> default	13:02:30.595976 -0400	Terraria.bin.osx	2   libdispatch.dylib                   0x00007fff900bd8b9 dispatch_once_f + 38
> default	13:02:30.595992 -0400	Terraria.bin.osx	3   CarbonCore                          0x00007fff7b7fd292 _Gestalt_SystemVersion + 976
> default	13:02:30.596008 -0400	Terraria.bin.osx	4   CarbonCore                          0x00007fff7b7fcaad Gestalt + 139
> default	13:02:30.596023 -0400	Terraria.bin.osx	5   libmojoshader.dylib                 0x000000012522c978 load_extensions + 1800
> default	13:02:30.596039 -0400	Terraria.bin.osx	6   libmojoshader.dylib                 0x000000012522c1b7 MOJOSHADER_glAvailableProfiles + 151
> default	13:02:31.535143 -0400	configd	LINKLOCAL en5: publish success
> default	13:02:31.583314 -0400	WindowServer	UI updates were forcibly disabled by application "<private>" for over 1.00 seconds. Server has re-enabled them.
> default	13:02:38.029519 -0400	WindowServer	UI updates were finally reenabled by application "<private>" after 7.45 seconds [0.13fps] (server forcibly re-enabled them after 1.00 seconds [1.00fps])
> default	13:02:38.107785 -0400	opendirectoryd	Client: <private>, UID: 0, EUID: 0, GID: 0, EGID: 0
> default	13:02:38.137340 -0400	opendirectoryd	Client: <private>, UID: 501, EUID: 501, GID: 20, EGID: 20
> default	13:02:38.233934 -0400	taskgated	MacOS error: -67062
> default	13:02:38.260105 -0400	opendirectoryd	Client: <private>, UID: 501, EUID: 501, GID: 20, EGID: 20
> default	13:02:38.278329 -0400	Steam Helper	[0929/130238.277856:WARNING:crash_reporting.cc(254)] Failed to set crash key: UserID with value: 0
> default	13:02:38.278437 -0400	Steam Helper	[0929/130238.278408:WARNING:crash_reporting.cc(254)] Failed to set crash key: BuildID with value: 1536409421
> default	13:02:38.278477 -0400	Steam Helper	[0929/130238.278458:WARNING:crash_reporting.cc(254)] Failed to set crash key: SteamUniverse with value: Public
> default	13:02:38.278512 -0400	Steam Helper	[0929/130238.278495:WARNING:crash_reporting.cc(254)] Failed to set crash key: Version with value: 1536409421
> default	13:02:38.278544 -0400	Steam Helper	[0929/130238.278528:WARNING:crash_reporting.cc(254)] Failed to set crash key: ProductName with value: cefwebhelper
> default	13:02:38.278576 -0400	Steam Helper	[0929/130238.278560:WARNING:crash_reporting.cc(254)] Failed to set crash key: Vendor with value: Valve
> default	13:02:38.324708 -0400	Steam Helper	Failed to look up the port for "com.apple.windowserver.active" (1100)
> default	13:02:43.699579 -0400	gamed	GKClientProxy: clientForBundleID:
> default	13:02:43.699724 -0400	gamed	GKClientProxy: updateIfRecentlyInstalled
> default	13:02:43.700267 -0400	gamed	GKClientProxy: clientForBundleID:
> default	13:02:43.700289 -0400	gamed	GKClientProxy: updateIfRecentlyInstalled

although some of those between the terraria.osx.bin and Steam Helper may be unrelated

**Comment by DrNorris** [comment]
Not sure if this will help, but if I totally reinstall Terraria and tmodloader, I do not hang on the black screen after the Re-Logic logo the first time I open Terraria.  Once I quit and start again that it will consistently hang on the black screen every other time I load the game.

My console output is as follows after which it will hang until I kill the process:

> AL lib: (WW) GetProcPath: Failed to readlink /proc/self/file: No such file or directory
> AL lib: (WW) alc_initconfig: Failed to initialize backend "jack"
> ﻿﻿﻿Setting breakpad minidump AppID = 105600
> Steam_SetMinidumpSteamID:  Caching Steam ID:  76561198019036587 [API loaded no]
> IGLDevice: OpenGLDevice
> OpenGL Device: NVIDIA GeForce GT 750M OpenGL Engine
> OpenGL Driver: 2.1 NVIDIA-12.0.19 355.11.10.50.10.103
> OpenGL Vendor: NVIDIA Corporation
> MojoShader Profile: glsl120
> Resolution changed to: 1920x1200.

**Comment by ClawsForHands** [comment]
Thank you so much!


________________________________
